infra/src/main/kotlin/fi/fta/geoviite/infra/publication/PublicationGeometryChangeRemarksUpdateService.kt

```diff
@@ -79,85 +77,80 @@ class PublicationGeometryChangeRemarksUpdateService(
-            val changedPoints =
-                oldPoints
-                    .mapIndexed { index, oldPoint -> index to oldPoint }
-                    .parallelStream()
-                    .map { (index, oldPoint) ->
-                        geocodingContext.getAddressAndM(oldPoint)?.let { (address, mOnReferenceLine) ->
-                            geocodingContext
-                                .getTrackLocation(newAlignment, address)
-                                ?.let { newAddressPoint ->
-                                    ComparisonPoints(mOnReferenceLine, index, oldPoint, newAddressPoint.point)
-                                }
-                                ?.let { comparison ->
-                                    if (comparison.roughDistance < changeThreshold) null else comparison
-                                }
-                        }
-                    }
-                    .toList()
-                    .filterNotNull()
-            val changedPointsRangesFirstIndices =
-                changedPoints
-                    .zipWithNext { a, b ->
-                        b.mOnReferenceLine - a.mOnReferenceLine >
-                            MINIMUM_M_DISTANCE_SEPARATING_ALIGNMENT_CHANGE_SUMMARIES
-                    }
-                    .mapIndexedNotNull { index, jump -> (index + 1).takeIf { jump } }
-            val changedPointsRanges =
-                (listOf(0) + changedPointsRangesFirstIndices + changedPoints.size).zipWithNext { a, b -> a to b }
-
-            if (changedPoints.isEmpty()) null
-            else
-                changedPointsRanges.mapNotNull { (from, to) ->
-                    val start = changedPoints[from]
-                    val end = changedPoints[to - 1]
-                    val startAddress = geocodingContext.getAddress(oldPoints[start.oldPointIndex])?.first
-                    val endAddress = geocodingContext.getAddress(oldPoints[end.oldPointIndex])?.first
-
-                    if (startAddress == null || endAddress == null) null
-                    else
-                        GeometryChangeSummary(
-                            end.mOnReferenceLine - start.mOnReferenceLine,
-                            changedPoints.subList(from, to).maxByOrNull { it.roughDistance }?.distance() ?: 0.0,
-                            startAddress,
-                            endAddress,
-                        )
-                }
-        }
+): List<GeometryChangeSummary> =
+    if (geometriesEqual(oldAlignment, newAlignment)) listOf()
+    else
+        getCommonAddressRange(geocodingContext, oldAlignment, newAlignment)?.let { (oldPoints, newPoints) ->
+            summarizeCommonAddressRange(geocodingContext, oldPoints, newPoints, changeThreshold)
+        } ?: listOf()
+
+private fun geometriesEqual(old: LayoutAlignment, new: LayoutAlignment): Boolean =
+    old.segments.map { it.geometry.id } == new.segments.map { it.geometry.id }
+
+private fun getCommonAddressRange(
```

Comment:
Näitä funkkareita katsellessa tuli mieleen toisessa kontekstissa tehty GeometryChangeCalculation.kt. Se ei tee ihan samaa, koska vertailee mikä geometriassa on muuttunut (ei osoitteissa) mutta jotain sukua nämä algoritmit selvästi on.